### PR TITLE
OpenSSL 3.0.8 updates & rebuilds, plus update PAGER logic, pagers...

### PR DESCRIPTION
**Pull request by satmandu**
- The OpenSSL 3.0.8 package includes the libraries from OpenSSL 1.1.1t, so existing packages do not break.
- Mold rebuild has the MOLD_JOBS variable only set for systems with less than 4Gb of free RAM.
- Libraries are set to only build as shared, and instructions to build static libraries have been removed.
- Updated `less` and `most`.
- `crew_profile_base` now asks for `PAGER` selection in the postinstall.
- `crew` is adjusted to install openssl first in a set of upgrades. (Really though, the upgrades should walk a dependency hierarchy to figure this order out automatically.)
- All relevant packages in a default install have been updated.
- The following code will show what other packages need updating:
```
for i in *.so ; do lddtree $i 2>/dev/null | grep "libcrypto.so.1.1\|libssl.so.1.1" && crew whatprovides $i ; done | grep -v "Total\|=" | awk -F : '{print $1}' | sort -u
``` 
Here's a list of packages on my system that are not broken but need to be rebuilt against OpenSSL 3.x based on running that command:
```
audacious_plugins
bind
cups
ffmpeg
freerdp
freetds
ghostscript
graphicsmagick
gstreamer
libevent
libgit2
libgxps
libimobiledevice
libkmod
libspectre
libssh2
libvncserver
mpv
neon
qtwebengine
spice_gtk
srt
tesseract
```

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=openssl308 CREW_TESTING=1 crew update ; crew upgrade ; crew upgrade

```

### CODE REVIEW

**Comment by satmandu** [comment]
I would love to have a way to do this in an interactive manner for a set of important packages, i.e. `ruby, openssl`
```
  # Only upgrade openssl if openssl is in the upgrade list, as other
  # package upgrades may break until crew is rerun with the new
  # version of openssl.
  if to_be_upgraded.include?('openssl')
    to_be_upgraded = ['openssl']
    @rerun_upgrade = true
  end
```

**Comment by supechicken** [comment]
No need as the OpenSSL shared library are loaded into the memory already before `ruby` runs. (which means it will not break the running `ruby` after the library was updated)

**Comment by satmandu** [comment]
> No need as the OpenSSL shared library are loaded into the memory already before `ruby` runs. (which means it will not break the running `ruby` process whatever the library is being deleted/updated)

The rationale for loading OpenSSL independently (well, first) is so that other packages (built against the newer OpenSSL) don't break during postinstalls which require the newer OpenSSL.

Maybe I should adjust this to just pop OpenSSL to the front of the install queue.

**Comment by satmandu** [comment]
Adjusted the `crew` logic to just add some code to make `openssl` install first during an upgrade.

File: packages/openssl.rb [hunk 1]
```diff
@@ -47,15 +44,18 @@ class Openssl < Package
   @ARCH_CXX_LTO_FLAGS = "#{@arch_cxx_flags} -flto=auto"
 
   def self.build
+    @no_tests_target = `openssl version | awk '{print $2}'`.chomp == version.to_s ? 'no-tests' : ''
+    puts "@no_tests_target is #{@no_tests_target}"
```
Comment (uberhacker) [inline packages/openssl.rb:48]:
```
puts "@no_tests_target is #{@no_tests_target}" unless @no_tests_target == ''
```
?

File: packages/openssl.rb [hunk 2]
```diff
@@ -65,10 +65,32 @@ def self.check
...
   end
 
   def self.install
     system "make DESTDIR=#{CREW_DEST_DIR} install_sw install_ssldirs"
+    # Extract OpenSSL 1.1.1t libraries for backwards compatibility purposes.
+    # Builds and rebuilds of packages against OpenSSL should automatically
+    # build against OpenSSL 3.x and not against OpenSSL 1.1.1x.
+    File.write 'openssl111_files', <<~EOF
+      usr/local/#{ARCH_LIB}/libcrypto.so.1.1
+      usr/local/#{ARCH_LIB}/libssl.so.1.1
```
Comment (uberhacker) [inline packages/openssl.rb:78]: What happens if `CREW_PREFIX` is something other than `/usr/local`?

File: packages/openssl.rb [hunk 3]
```diff
@@ -65,10 +65,32 @@ def self.check
...
+    case ARCH
+    when 'aarch64', 'armv7l'
+      downloader 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/1.1.1t_armv7l/openssl-1.1.1t-chromeos-armv7l.tar.xz',
+                 '88a36d1539c7c01af1f5e469b64c2760f43126bb75c0e63b53d3d61c2a6fbe7f', 'openssl-1.1.1t-chromeos.tar.xz'
+    when 'i686'
+      downloader 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/1.1.1t_i686/openssl-1.1.1t-chromeos-i686.tar.xz',
+                 'ce98c1898e57df1cbcceab08912e219fb5f27b0e4585315f7babdf524fa844dc', 'openssl-1.1.1t-chromeos.tar.xz'
+    when 'x86_64'
+      downloader 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/1.1.1t_x86_64/openssl-1.1.1t-chromeos-x86_64.tar.xz',
+                 'd6583dc2c7566da33402cb8d7f9025189d65b8ba6300e752edcb0d74ebcb1f68', 'openssl-1.1.1t-chromeos.tar.xz'
```
Comment (uberhacker) [inline packages/openssl.rb:90]: There are a lot of redundant versions here.  Would it make sense to use `#{version}`?

File: packages/openssl.rb (hunk 2 above)
Comment (satmandu) [reply packages/openssl.rb:78]: I think we have torn out the plumbing to enable that in any case... but open to code to just remove the first slash from CREW_PREFIX...

File: packages/openssl.rb (hunk 3 above)
Comment (satmandu) [reply packages/openssl.rb:90]: Code suggestion to replace this block?

File: packages/openssl.rb (hunk 2 above)
Comment (uberhacker) [reply packages/openssl.rb:77]: This is definitely an example of tearing out the plumbing. ;)

File: packages/openssl.rb (hunk 3 above)
Comment (uberhacker) [reply packages/openssl.rb:89]:
```
    @legacy_version = '1.1.1t'
    case ARCH
    when 'aarch64', 'armv7l'
      downloader "https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/#{@legacy_version}_armv7l/openssl-#{@legacy_version}-chromeos-armv7l.tar.xz",
                 '88a36d1539c7c01af1f5e469b64c2760f43126bb75c0e63b53d3d61c2a6fbe7f', "openssl-#{@legacy_version}-chromeos.tar.xz"
    when 'i686'
      downloader "https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/#{@legacy_version}_i686/openssl-#{@legacy_version}-chromeos-i686.tar.xz",
                 'ce98c1898e57df1cbcceab08912e219fb5f27b0e4585315f7babdf524fa844dc', "openssl-#{@legacy_version}-chromeos.tar.xz"
    when 'x86_64'
      downloader "https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/#{@legacy_version}_x86_64/openssl-#{@legacy_version}-chromeos-x86_64.tar.xz",
                 'd6583dc2c7566da33402cb8d7f9025189d65b8ba6300e752edcb0d74ebcb1f68', "openssl-#{@legacy_version}-chromeos.tar.xz"
...
system "tar -xv --files-from #{@cur_dir}/openssl111_files -f #{@cur_dir}/openssl-#{@legacy_version}-chromeos.tar.xz"
```

File: packages/openssl.rb (hunk 2 above)
Comment (satmandu) [reply packages/openssl.rb:77]: `CREW_PREFIX.chr` might fix this...

File: packages/openssl.rb (hunk 3 above)
Comment (satmandu) [reply packages/openssl.rb]: Done.

File: packages/crew_profile_base.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ class Crew_profile_base < Package
   description 'Crew-profile-base sets up Chromebrew\'s environment capabilities.'
   homepage 'https://github.com/chromebrew/crew-profile-base'
   @_ver = '0.0.8'
-  version @_ver
+  version "#{@_ver}-1"
```
Comment (uberhacker) [inline packages/crew_profile_base.rb:7]: Why not just increment from 0.0.8 to 0.0.9?  This is our package.

**Review by uberhacker — APPROVED** [review]
Nice job.  I couldn't find anything wrong with these updates.

**Pull request merged**